content/ADR-204-comms-architecture.md

```diff
@@ -0,0 +1,203 @@
+	C->>WS: signedChallenge  
+	WS->>WS: validateSignature 
+	WS-->>C: Welcome
+end	
+```
+
+#### 2. Archipelago Island 
+
+
+```mermaid
+sequenceDiagram
+participant C as Decentraland Client
+participant R as Realm Provider
+participant WS as WebSocket Connector
+participant N as NATS
+participant A as Archipelago Core
+participant L as LiveKit 
+Note over A: Initialize LiveKit Transport
+  A->>N: subscribe: peer.*.heartbeat/disconnect
+  C->>WS: Heartbeat with position 
+	WS->>N: publish: peer.0x....heartbeat
+	WS->>N: subscribe: peer.0x...island_changed 
+	A->>N: publish: peer.0x...island_change: islandId & connStr 
+	N-->>WS: islandID & connectionStr
+	WS-->>C: IslandId & connectionStr
+	C->>L: connect to LiveKit room Id = islandId
+	L-->>C: roomConnection
+	C->>L: msgs to roomId: movement / nearby chat
+	L-->>L: brocast msg to roomId participants 
+```
+
+#### 3. Scene Room
+
+```mermaid
+sequenceDiagram
+actor User
+participant C as Decentraland Client
+participant G as GateKeeper 
+participant L as LiveKit 
+
+  User->>C: move to parcel 
+	C->>G: Get sceneRoom 
+	G-->>C: room connectionStr
+	C->>L: Connect to Scene Room (realm:sceneId): connectionStr
+	L-->>C: roomConnection 
+	C->>L: send message: movement / playEmote / chat / voice / announceProfileV
+	L-->>L: brodcast msg to roomId participats 
+```
+
+
+## Protocol Messages  
+
+The protocol messages for profiles, positions, and updates have been optimized for client performance, aiming to minimize data transmission through the channels. In this new implementation, profiles are no longer sent through the island and are instead retrieved from the Catalyst network. When a user updates their profile, a notification is sent, and the updated profile is downloaded from the servers. This approach ensures that avatar impersonation is prevented, as the signed profile is retrieved from a trusted Catalyst node.
+
+TBD 
+
+## Deadline
+
+Date: TBD
+
+## Consequences
+
+- The protocol and architecture for the communication services will undergo significant changes with this implementation. This means that users connecting through clients built on the previous version of the protocol will not be able to interact with those using clients on this new version. 
+- Voice chat will be routed through the scene room. If one user is on a road and another is on the floor, each in different scenes, they won’t be able to hear each other. This decision was made to give the scene owners the control over the voice chat on their scenes. 
+- Duplicated messages, such as positions, profiles, and text chats, may be transmitted across both island and scene rooms. This can lead to inefficient data transfer; however, it can be optimized on the client side by adjusting the message frequency for each channel based on the number of active connections. This design trade-off was necessary to meet all criteria and to visualize and interact with users within the scene and users close to the scene boundaries.
```

Comment:
```suggestion
- Duplicated messages, such as positions, profiles, and text chats, may be transmitted across both island and scene rooms. This can lead to inefficient data transfer; however, it can be optimized on the client side by adjusting the message frequency for each channel based on the number of active connections. This design trade-off was necessary to meet all the criteria and enable visualization and interaction with users both within the scene and near its boundaries.
```